### PR TITLE
feat: Add example helm chart

### LINKED ISSUE #87: Add example helm chart
**Is your feature request related to a problem? Please describe.**

Many people choose to use helm charts when deploying to k8s; we even use one in internal dogfooding!

**Describe the solution you'd like**

Add a simple helm chart in this repo. Later we can move it to a dedicated repo for easier releasing, etc.

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Adds a example helm chart that can be used to install and configure the agent.

- Closes #87 

## Short description of the changes
- Adds example helm chart with values and template files
- Reference the helm chart in Developing markdown

### CODE REVIEW

**Comment by vreynolds** [comment]
@TylerHelmuth can we get your eyes on this? We're hoping to iron out helm things in this example before publishing an actual chart. 🙏

File: DEVELOPING.md [hunk 1]
```diff
@@ -14,6 +14,21 @@ Recommended:
   - View them with `remake --tasks`
 - [locust](https://docs.locust.io/en/stable/what-is-locust.html) - A performance testing tool, required for load testing
 
+## Installing with Helm
+
+We have an example Helm chart that can be used to configure and run the agent.
+Update [values.yaml](./chart/values.yaml) with your Honeycomb API key then run:
+
+```sh
+helm install -f chart/values.yaml hny-network-agent ./chart
```
Comment (TylerHelmuth) [inline DEVELOPING.md:23]:
```suggestion
helm install hny-network-agent ./chart
```

File: chart/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+name: hny-network-agent
+description: Honeycomb Network Agent
+version: 0.0.1
+appVersion: v0.0.16-alpha
+home: https://honeycomb.io
+sources:
+  - https://github.com/honeycombio/honeycomb-network-agent
+icon: https://www.honeycomb.io/wp-content/themes/honeycomb/assets/img/logo.svg
```
Comment (TylerHelmuth) [inline chart/Chart.yaml:9]:
```suggestion
icon: https://www.honeycomb.io/wp-content/themes/honeycomb/assets/img/logo.svg

```

File: chart/templates/template.yaml
Comment (TylerHelmuth) [inline chart/templates/template.yaml:1]: I'd suggest splitting each object into is own file.

File: DEVELOPING.md (hunk 1 above)
Comment (MikeGoldsmith) [reply DEVELOPING.md:23]: The main use-case we can see for a values file would be to pass in the Honeycomb API key and maybe replace the API endpoint events are sent to. If we don't use the `-f` flag here, how would you recommend to do that instead?

Comment (TylerHelmuth) [reply DEVELOPING.md:23]: the way it was written it was passing the chart's default values.yaml, but that is equivalent to not providing a `-f` or `--values` flag.

Comment (TylerHelmuth) [reply DEVELOPING.md:23]: Users should provide their own values.yaml file, or use the `--set` flag, to override the default values.yaml.

**Review by TylerHelmuth — COMMENTED** [review]
I recommend adding a NOTES.txt file that will fail if an API key has not been set.  You can see a similar example here: https://github.com/honeycombio/helm-charts/blob/main/charts/honeycomb/templates/NOTES.txt

File: chart/templates/template.yaml
Comment (MikeGoldsmith) [reply chart/templates/template.yaml:1]: I've broken out the files as suggested in https://github.com/honeycombio/honeycomb-network-agent/pull/221/commits/3fa73c414f15aea2fa9f0c8ffed1aab2beea279d 👍🏻

**Comment by MikeGoldsmith** [comment]
Closing in favour of publishing the chart here instead:
- https://github.com/honeycombio/helm-charts/pull/292

**Pull request closed without merging**